### DISCUSSION TITLE
Windows TLS1.3 and System.ComponentModel.Win32Exception (0x80090326)

### DISCUSSION THREAD

**Discussion by MisterErwin** [ppy/osu#27568] [Q&A] [answered]
### Type

Other

### Bug description

When trying to log in (or download the initial set of beat maps during set up),
one is greeted by an (untranslated) "An error occurred while sending the request.".

The interesting lines in the log should be the following (system language is set to German):

> 2024-03-10 22:35:18 [verbose]: Failed to update: System.Net.Http.HttpRequestException: An error occurred while sending the request.
2024-03-10 22:35:18 [verbose]: ---> System.IO.IOException: The read operation failed, see inner exception.
2024-03-10 22:35:18 [verbose]: ---> System.Security.Authentication.AuthenticationException: Authentication failed, see inner exception.
2024-03-10 22:35:18 [verbose]: ---> System.ComponentModel.Win32Exception (0x80090326): Das Format der empfangenen Nachricht war unerwartet oder fehlerhaft.
2024-03-10 22:35:18 [verbose]: --- End of inner exception stack trace ---
2024-03-10 22:35:18 [verbose]: at System.Net.Security.SslStream.ForceAuthenticationAsync[TIOAdapter](Boolean receiveFirst, Byte[] reAuthenticationData, CancellationToken cancellationToken)
2024-03-10 22:35:18 [verbose]: at System.Net.Security.SslStream.ReplyOnReAuthenticationAsync[TIOAdapter](Byte[] buffer, CancellationToken cancellationToken)
2024-03-10 22:35:18 [verbose]: at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](Memory`1 buffer, CancellationToken cancellationToken)
2024-03-10 22:35:18 [verbose]: --- End of inner exception stack trace ---

A wireshark  dump indicates TLS 1.3 is used (as does the registry entry `Computer\HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.3\Client` `Enabled=1`)

When the registry entry  is changed to `Enabled=0`, the login (& everything else) works again.

Forcing TLS 1.2 to be used might be an option (cf. https://github.com/dotnet/runtime/issues/91464)

Thanks :)

### Screenshots or videos

![login error](https://github.com/ppy/osu/assets/1916723/9051ac0a-a1d4-4e1f-ba35-2878950668c3)

![wireshark](https://github.com/ppy/osu/assets/1916723/6876162f-25c2-4def-b6ed-96855f63b541)

### Version

2024.302.1-lazer

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/14552033/compressed-logs.zip)

**Comment by bdach** [comment, marked as answer]
I'm downgrading this to a discussion because I cannot reproduce any of this.

It's not TLS 1.3. My game instance is using 1.3 perfectly fine.

![image](https://github.com/ppy/osu/assets/20418176/2360edc7-d9c4-49f1-aef2-62d46659e116)
![image](https://github.com/ppy/osu/assets/20418176/cd1b8c6c-bec8-42cc-8bd6-bd87331aeae7)

> (as does the registry entry `Computer\HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.3\Client` `Enabled=1`)

I don't even _have_ this registry entry.

> ![wireshark](https://private-user-images.githubusercontent.com/1916723/311550599-6876162f-25c2-4def-b6ed-96855f63b541.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAxNDY2MTUsIm5iZiI6MTcxMDE0NjMxNSwicGF0aCI6Ii8xOTE2NzIzLzMxMTU1MDU5OS02ODc2MTYyZi0yNWMyLTRkZWYtYjZlZC05Njg1NWY2M2I1NDEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMTFUMDgzODM1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTg4MDkxYjhkNmU0NzI2M2VlMzliZGYyMTlmOGU3OWY3N2JjZTY0ZTcwNjA1N2U2NmEwYWQyZWRlZWFiYmY3NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Re2BwXNDHk11XxHKioKGs2aXXUuPgR669PHT0DDBJoE)

I don't get how this is using the ipv6 address. I suspect that if anything this is probably going to be related closer to that, but I can't get my windows to use it. And it should be already preferring ipv6 as per [docs](https://learn.microsoft.com/en-US/troubleshoot/windows-server/networking/configure-ipv6-in-windows) because I have `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Tcpip6\Parameters\` unset, which means that it should default to `0x00`, which means it should be preferring ipv6 over ipv4 (bit 5 unset). And yet it doesn't.

@MisterErwin how did you get to this state of the registry even? Did you manually edit registry entries / alter your windows configuration in a way that might cause this?

@ppy/team-web @ThePooN maybe you can have a look at what might be going on here? I don't think we should be patching the client of all things for this.

**Reply by MisterErwin** [reply]
@bdach Thanks for looking into this too.

After also trying to replicate it on a fresh install of windows: I have no idea what has caused my problem.

My vague guess would be some windows IIS server experiments on my machine, but I was unable to replicate the steps.

(Switching between IPv6/IPv4, different DNS servers, various windows feature packs did not change anything on the problematic machine)

(I guess this discussion can now be useful for some poor soul running into the same problem - hey me in two years)

TL;DR: I messed up my windows install somewhat - changing/deleting the registry entry appears to "fix" it.

**Reply by ThePooN** [reply]
i can't make sense of this either.
